src/features/users/usersSlice.ts

```diff
@@ -0,0 +1,27 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { User } from '../../types/User';
+import { getUsers } from '../../api/users';
+
+export const fetchUsers = createAsyncThunk<User[]>(
+  'users/fetchUsers',
+  async () => {
+    const response = getUsers();
```

Comment:
It seems like you're directly returning the result of `getUsers()` without awaiting the promise. As `getUsers` is an asynchronous operation, you should await its result before returning to ensure that the thunk resolves with the actual users data.